### PR TITLE
Graduate PodDeletionCost to Beta

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Graduates PodDeletionCost feature to Beta. 

#### Which issue(s) this PR fixes:
Part of #45509 https://github.com/kubernetes/enhancements/issues/2255

#### Does this PR introduce a user-facing change?

```release-note
PodDeletionCost promoted to Beta
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2255-pod-cost
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -858,7 +859,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	PreferNominatedNode:                            {Default: false, PreRelease: featuregate.Alpha},
 	ProbeTerminationGracePeriod:                    {Default: false, PreRelease: featuregate.Alpha},
 	RunAsGroup:                                     {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.22
-	PodDeletionCost:                                {Default: false, PreRelease: featuregate.Alpha},
+	PodDeletionCost:                                {Default: false, PreRelease: featuregate.Beta},
```
Comment (janetkuo) [inline pkg/features/kube_features.go:862]: Beta features should be enabled by default

Comment (ahg-g) [reply pkg/features/kube_features.go]: oops

Comment (ahg-g) [reply pkg/features/kube_features.go]: done.

**Comment by liggitt** [comment]
is there an e2e test of this feature that should be enabled by default now that this is beta?

**Comment by liggitt** [comment]
doc update API change lgtm

**Comment by ahg-g** [comment]
> is there an e2e test of this feature that should be enabled by default now that this is beta?

No, there is an integration test only.

**Comment by liggitt** [comment]
ok

/approve

**Review by janetkuo — APPROVED** [review]

**Comment by remiville** [comment]
Hello @ahg-g ,
Thanks for your work !
I'm using PodDeletionCost to adapt on on-premise worker pool to k8s.
Previously the product was using custom metrics to do so, but it has drawbacks.

I know the feature is BETA and not planned for GA.
From https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/#avoiding-permanent-beta I'm not sure if there's a risk this feature will be automatically removed in a future k8s version ?
I'm not sure if this feature is considered as a feature's API or not ?

I would except this feature to be kept in future k8s release version until an improvement or a better proposal is implemented that can be used instead.
Will it be the case ?

**Comment by liggitt** [comment]
https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/#avoiding-permanent-beta currently is limited in scope to beta API *versions* (e.g. `v1beta1`), so does not currently apply to specific fields or feature gates. That said, a feature gate living in beta for an extended period of time is not desired, since it attracts usage without making official stability guarantees.

**Pull request merged**